File: cs/ch03/Money.Tests/DollarTest.cs
using NUnit.Framework;
using TheSoftwareGorilla.TDD.Money;

namespace TheSoftwareGorilla.TDD.Money.Tests;


#region TODO List
//TODO: $5 + 10 CHF = $10 if rate is 2:1
//TODO: $5 * 2 = $10 - DONE 
//TODO: Make "amount" private
//TODO: Dollar side-effects? - DONE
//TODO: Money rounding?
//TODO: equals() -DONE
//TODO: hashCode()
//TODO: Equal null
//TODO: Equal object
#endregion

public class DollarTests
{
    
    [SetUp]
    public void Setup()
    {
        // No setup required for these tests
    }

    [Test]
    public void TestConstruction()
    {
        Dollar five = new Dollar(5);
        Assert.That(five, Is.Not.Null);
        Assert.That(five.Amount, Is.EqualTo(5));
        var ten = new Dollar(10);
        Assert.That(ten, Is.Not.Null);
        Assert.That(ten.Amount, Is.EqualTo(10));
    }

    [Test]
    public void TestMultiplication()
    {
        Dollar five = new Dollar(5);
        Dollar product = five.Times(2);
        Assert.That(product.Amount, Is.EqualTo(10));
        product = five.Times(3);
        Assert.That(product.Amount, Is.EqualTo(15));
    }

    [Test]
    public void TestEquality()
    {
        Assert.That(new Dollar(5), Is.EqualTo(new Dollar(5)));
        Assert.That(new Dollar(5), Is.Not.EqualTo(new Dollar(6)));
    }
}

File: java/ch03/src/test/java/com/thesoftwaregorilla/tdd/money/DollarTest.java
package com.thesoftwaregorilla.tdd.money;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

//<editor-fold desc="TO DO List">
//TODO: $5 + 10 CHF = $10 if rate is 2:1
//TODO: $5 * 2 = $10 - DONE 
//TODO: Make "amount" private
//TODO: Dollar side-effects? - DONE
//TODO: Money rounding?
//TODO: equals() -DONE
//TODO: hashCode()
//TODO: Equal null
//TODO: Equal object
//</editor-fold>

public class DollarTest {

    @Test
    public void testConstruction() {
        Dollar five = new Dollar(5);
        assertEquals(5, five.amount);
        Dollar ten = new Dollar(10);
        assertEquals(10, ten.amount);
    }

    @Test
    public void testMultiplication() {
        Dollar five = new Dollar(5);
        Dollar product = five.times(2);
        assertEquals(10, product.amount);
        product = five.times(3);
        assertEquals(15, product.amount);
    }

    @Test
    public void testEquality() {
        assertEquals(new Dollar(5), new Dollar(5));
        assertNotEquals(new Dollar(5), new Dollar(6));
    }

}
